From ecdbdb5039dd7f016918f8d734a11670ea062bf7 Mon Sep 17 00:00:00 2001
From: Benjamin Berg <bberg@redhat.com>
Date: Sun, 6 Feb 2022 19:24:57 +0100
Subject: [PATCH 5/7] adaptive: Open event FD non-blocking and with CLOEXEC
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

This makes more sense, and it is what libevdev expects by default.
---
 src/thd_engine_adaptive.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/thd_engine_adaptive.cpp b/src/thd_engine_adaptive.cpp
index 232dcd4..c546b61 100644
--- a/src/thd_engine_adaptive.cpp
+++ b/src/thd_engine_adaptive.cpp
@@ -1550,7 +1550,7 @@ void cthd_engine_adaptive::setup_input_devices() {
 		int fd = -1;
 
 		snprintf(fname, sizeof(fname), "/dev/input/%s", namelist[i]->d_name);
-		fd = open(fname, O_RDONLY);
+		fd = open(fname, O_RDONLY | O_NONBLOCK | O_CLOEXEC);
 		if (fd < 0)
 			continue;
 		ret = libevdev_new_from_fd(fd, &tablet_dev);
-- 
2.35.1

